libavcodec/atrac3plusdec.c

```diff
@@ -339,24 +339,38 @@ static int atrac3p_decode_frame(AVCodecContext *avctx, void *data,
     if ((ret = ff_get_buffer(avctx, frame, 0)) < 0)
         return ret;
 
-    if ((ret = init_get_bits8(&ctx->gb, avpkt->data, avpkt->size)) < 0)
+    // PPSSPP workaround: With bad/corrupt input, the atrac3plus decoder does not
+    // reliably stay inside the bounds of the buffer. Instead of carefully checking everything
+    // inside it, for now let's just give it more space to read from.
+    const int extra_bytes = 1024;
+
+    uint8_t *bigger_buffer = malloc(avpkt->size + extra_bytes);
+    memset(bigger_buffer + avpkt->size, 0, extra_bytes);
+    memcpy(bigger_buffer, avpkt->data, avpkt->size);
```

Comment:
Is this directly the packet buffer we pass in via avcodec_decode_audio4?  If so, we can control the buffer size there.  Usually it points at PSP RAM, though... but it can also point to `dataBuf_`.  Does padding that out help?

-[Unknown]

Reply:
Oh, I was so happy to have found the workaround that I forgot to track outwards to see if we control the buffer. Will have a look later today.